File: Data Structures & Algorithms/1894-MergeStringsAlternately/1894-MergeStringsAlternately.cpp
// Last updated: 31/05/2026, 14:02:37
class Solution {
public:
    string mergeAlternately(string word1, string word2) {
        string res = "";
        int i = 0;
        int j = 0;
        while(i < word1.size() || j < word2.size()){
            if(i != word1.size()){
                res.push_back(word1[i]);
                i++;
            }
            if(j != word2.size()){
                res.push_back(word2[j]);
                j++;
            }
        }
        return res;
    }
};

File: Data Structures & Algorithms/1894-MergeStringsAlternately/1894-MergeStringsAlternately.java
// Last updated: 07/07/2026, 09:09:14
class Solution {
    public String mergeAlternately(String word1, String word2) {
        String res = "";
        int i = 0;
        int j = 0;
        while(i!=word1.length() || j!=word2.length()){
            if(i!=word1.length()){
                res = res + word1.charAt(i);
                i++;
            }
            if(j!=word2.length()){
                res = res + word2.charAt(j);
                j++;
            }
        }
        return res;
    }
}